extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/services.repositories.js

```diff
@@ -148,7 +148,11 @@ angular.module('PaperUI.services.repositories', []).factory('bindingRepository',
             repository.add(discoveryResult);
         }
         if (topic.indexOf("removed") > -1 && index != -1) {
-            repository.remove(discoveryResult, index);
+            repository.remove(undefined, index);
+        }
+        if (topic.indexOf("updated") > -1 && index != -1) {
+            repository.remove(undefined, index);
+            repository.add(discoveryResult);
```

Comment:
Just a blind guess - is the order relevant in any way?

Reply:
No, not relevant because `add` will append to the data array, `remove` will use the previously calculated index (by thingUID).

Reply:
Now I got it. The UI orders the discovery results by label during rendering. So order in the repository does not matter.